nautobot/extras/datasources/git.py

```diff
@@ -155,7 +155,7 @@ def ensure_git_repository(repository_record, logger=None, head=None):  # pylint:
         with suppress(InvalidGitRepositoryError):
             if (
                 Path(repository_record.filesystem_path).exists()
-                and Repo(repository_record.filesystem_path).rev_parse("HEAD") == head
+                and str(Repo(repository_record.filesystem_path).rev_parse("HEAD")) == head
```

Comment:
Maybe, just to be extra extra sure?

```suggestion
                and str(Repo(repository_record.filesystem_path).rev_parse("HEAD")) == str(head)
```

Reply:
Done on both. I like it!